include/ada/url_search_params.h

```diff
@@ -0,0 +1,75 @@
+/**
+ * @file url_search_params.h
+ * @brief Declaration for the URL Search Params
+ */
+#ifndef ADA_URL_SEARCH_PARAMS_H
+#define ADA_URL_SEARCH_PARAMS_H
+
+namespace ada {
+
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+/**
+ * @see https://url.spec.whatwg.org/#interface-urlsearchparams
+ */
+struct url_search_params {
+  // TODO: Add missing constructors
+  url_search_params() = default;
+  url_search_params(const url_search_params &u) = default;
+  url_search_params(url_search_params &&u) noexcept = default;
+  url_search_params &operator=(url_search_params &&u) noexcept = default;
+  url_search_params &operator=(const url_search_params &u) = default;
+  ~url_search_params() = default;
+
+  [[nodiscard]] inline size_t size() const noexcept;
+
+  /**
+   * @see https://url.spec.whatwg.org/#dom-urlsearchparams-append
+   */
+  inline void append(std::string_view key, std::string_view value);
+
+  /**
+   * @see https://url.spec.whatwg.org/#dom-urlsearchparams-delete
+   */
+  void remove(std::string_view key);
+  void remove(std::string_view key, std::string_view value);
+
+  /**
+   * @see https://url.spec.whatwg.org/#dom-urlsearchparams-get
+   */
+  inline std::optional<std::string_view> get(std::string_view key);
+
+  /**
+   * @see https://url.spec.whatwg.org/#dom-urlsearchparams-getall
+   */
+  inline std::vector<std::string> get_all(std::string_view key);
```

Comment:
The spec seems to require the return of all tuples, is that right?

Reply:
`get_all` returns values, such as:

```
➜  url-parser git:(url-search-params) ✗ node
Welcome to Node.js v20.3.0.
Type ".help" for more information.
> let params = new URLSearchParams('key=value&key=value2')
undefined
> params.getAll('key')
[ 'value', 'value2' ]
```